src/compiler/wasm-compiler.cc

```diff
@@ -4332,38 +4339,104 @@ class WasmWrapperGraphBuilder : public WasmGraphBuilder {
     }
   }
 
-  Node* FromJS(Node* node, Node* js_context, wasm::ValueType type) {
+  Node* BuildChangeInt64ToBigInt(Node* input) {
+    WasmToJavaScriptTypeConversionDescriptor interface_descriptor;
+
+    auto call_descriptor = Linkage::GetStubCallDescriptor(
+        mcgraph()->zone(),                              // zone
+        interface_descriptor,                           // descriptor
+        interface_descriptor.GetStackParameterCount(),  // stack parameter count
+        CallDescriptor::kNoFlags,                       // flags
+        Operator::kNoProperties,                        // properties
+        stub_mode_);                                    // stub call mode
+
+    /**
+     * When calling an imported func the conversion is initiated from wasm and
+     * the target code must be PIC.
+     *
+     * See more details in src/snapshot/snapshot-common.cc +388
+     */
+    auto index = wasm::WasmCode::kWasmToBigIntJavaScript_WasmRuntimeTrampoline;
+
+    Node* target = (stub_mode_ == StubCallMode::kCallWasmRuntimeStub)
+                       ? mcgraph()->RelocatableIntPtrConstant(
+                             index,
+                             RelocInfo::WASM_STUB_CALL)
+                       : jsgraph()->HeapConstant(
+                             BUILTIN_CODE(isolate_, WasmToBigIntJavaScript));
```

Comment:
Dumb question (I haven't worked with CSA before): Rather than calling this runtime stub, would it work to call CodeStubAssembler::BigIntFromInt64 directly, or write code that's similar to BuildChangeInt32ToTagged to do accomplish this task?

Reply:
Your question make sense and it would be possible in JS, wasm has a additionnal restriction for calling its runtime stubs, see https://github.com/xtuc/v8/pull/1/files/dda86c1f2fc8c7dfee946527154ae5cccffde6a4#diff-74de4cf1fe91d141c93cad45c7cb81e0R4357. It's more or less a workarround.

Reply:
Here's the comment https://github.com/v8/v8/blob/master/src/snapshot/snapshot-common.cc#L304-L307

Reply:
Hmm, is BuildChangeInt32ToTagged a runtime stub, or just a bunch of compiler IR that gets generated inline? I thought it was the latter, and that's what I'm suggesting be done here.

Reply:
`BuildChangeInt32ToTagged` generates IR, what you're suggesting is interesting, it could avoid the additional procedure call. I think that I could generate the IR directly too, thanks for your suggestion!

Reply:
I'm not sure how to do because I'd like to use the code in BigIntFromInt64 on the CSA. `BuildChangeInt32ToTagged` is a bit simplier and probably not existing already on the CSA.

Reply:
How I implemented is consistent with the other wasm buitilns or trampolines to builtin. Howerver, the code structure is wrong?

Reply:
I think it'd be OK to go down either of these paths, as BigInts have significant performance overhead anyway (though my personal preference would be the faster one--maybe you can work out how to share the code).

Reply:
I though about that also, the conversion only happen at the JS boudaries. In wasm it's a unboxed int64 type.

Howerver, I'd like to make it right first.